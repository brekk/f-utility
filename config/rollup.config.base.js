const progress = require(`rollup-plugin-progress`)
const babili = require(`rollup-plugin-babili`)
const commonjs = require(`rollup-plugin-commonjs`)
const cleanup = require(`rollup-plugin-cleanup`)
const resolve = require(`rollup-plugin-node-resolve`)
const buble = require(`rollup-plugin-buble`)
const json = require(`rollup-plugin-json`)
const pkg = require(`../package.json`)
const external = Object.keys(pkg.dependencies)

module.exports = {
  exports: `named`,
  external,
  globals: {
  },
  moduleName: pkg.name,
  plugins: [
    progress(),
    json(),
    commonjs({
      extensions: [`.js`],
      include: `node_modules/**`,
      namedExports: {
      }
    }),
    buble(),
    resolve({
      jsnext: true,
      main: true
    }),
    cleanup({
      comments: `none`
    }),
    babili({
      removeConsole: true
    })
  ]
}
