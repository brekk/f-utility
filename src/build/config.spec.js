import config from "./config"

test("config", () => {
  expect(config).toMatchSnapshot()
  expect(config.UNCHECKED.ts()).toEqual("any")
})
