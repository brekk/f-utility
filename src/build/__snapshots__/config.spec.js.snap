// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config 1`] = `
Object {
  "AUTO": Object {
    "check": true,
    "name": "@@FUTILITY::config.auto",
    "ts": [Function],
  },
  "CHECKED": Object {
    "check": true,
    "name": "@@FUTILITY::config.checked",
    "ts": [Function],
  },
  "UNCHECKED": Object {
    "check": false,
    "name": "@@FUTILITY::config.unchecked",
    "ts": [Function],
  },
}
`;
